tests/phpunit/tests/blocks/wpBlockMetadataRegistry.php

```
@@ -188,4 +189,35 @@ public function test_register_collection_with_non_existent_manifest() {
 		$result = WP_Block_Metadata_Registry::register_collection( '/var/arbitrary/path', $non_existent_manifest );
 		$this->assertFalse( $result, 'Non-existent manifest should not be registered' );
 	}
+
+	/**
+	 * Tests that the `get_collection_block_metadata_files()` method returns the expected list of block metadata files.
+	 *
+	 * @ticket 62267
+	 * @covers ::get_collection_block_metadata_files
+	 */
+	public function test_get_collection_block_metadata_files() {
+		$path          = WP_PLUGIN_DIR . '/test-plugin/data/block-types';
+		$manifest_data = array(
+			'a-block'       => array(
+				'name'  => 'a-block',
+				'title' => 'A Block',
+			),
+			'another-block' => array(
+				'name'  => 'another-block',
+				'title' => 'Another Block',
+			),
+		);
+
+		file_put_contents( $this->temp_manifest_file, '<?php return ' . var_export( $manifest_data, true ) . ';' );
+
+		$this->assertTrue( WP_Block_Metadata_Registry::register_collection( $path, $this->temp_manifest_file ) );
+		$this->assertSame(
+			array(
+				WP_PLUGIN_DIR . '/test-plugin/data/block-types/a-block/block.json',
+				WP_PLUGIN_DIR . '/test-plugin/data/block-types/another-block/block.json',
+			),
+			WP_Block_Metadata_Registry::get_collection_block_metadata_files( $path )
+		);
```

The test should also verify whether the block types were successfully registered. Something like:

```php
$this->assertTrue( WP_Block_Type_Registry::is_registered( 'a-block' ) );
```

By the way, it should fail during the registration at the moment because of the missing namespace in the `name`:

https://github.com/WordPress/wordpress-develop/blob/f89ea16aca46fdb809eb8a43a9a295d30800a08f/src/wp-includes/class-wp-block-type-registry.php#L73-L81

---

Ah wait, that would be in the test for `wp_register_block_types_from_metadata_collection` but it isn't covered yet.